assets/RNAseq_report.Rmd

```diff
@@ -152,14 +330,287 @@ if(file.exists(QAfile)){
     }
     df_QA_neat <- merge(df_QA_neat, df_QA_cond, by="row.names")
     df_QA_neat$Row.names <- NULL
-    if ("RIN" %in% colnames(df_QA)){
-        if (any(is.na(df_QA$RIN))){no_RIN <- TRUE} else {RIN <- TRUE; df_QA_neat$RIN <- df_QA$RIN}
-    } else {no_RIN <- TRUE}
+    if ("RIN" %in% colnames(df_QA)) {
+        if (any(is.na(df_QA$RIN))) {
+            no_RIN <- TRUE
+        } else {
+            RIN <- TRUE
+            df_QA_neat$RIN <- df_QA$RIN
+        }
+    } else {
+        no_RIN <- TRUE
+    }
 } else {
     no_RIN <- TRUE
 }
 ```
 
+<!-- DESeq2 start -->
+<!-- This block needs to be executed before the rest of the code as the variables will be needed -->
+
+```{r DESeq, echo=FALSE, message=FALSE, warning=FALSE, results = 'hide'}
+################## RUN DESEQ2 ######################################
+#Apply relevel if provided to metadata
+if (!isNot(params$path_relevel)) {
+    relevel_table <- read.table(params$path_relevel, sep="\t", header = T, colClasses = "character")
+    write.table(relevel_table, file="differential_gene_expression/metadata/relevel.tsv")
+    for (i in c(1:nrow(relevel_table))) {
+        relev <- relevel_table[i,]
+        cds[[paste(relev[1])]] <- relevel(cds[[paste(relev[1])]], paste(relev[2]))
+    }
+}
+# Run DESeq function
+if (params$input_type == "featurecounts") {
+    cds <- DESeqDataSetFromMatrix( countData =count.table, colData =metadata, design = eval(parse(text=as.character(design[[1]]))))
+    cds <- DESeq(cds,  parallel = FALSE)
+} else if (params$input_type %in% c("rsem", "salmon")) {
+    ## Create a dataframe which consists of both the gene id and the transcript name
+    gtf <- rtracklayer::import(params$path_gtf)
+    gtf <- as.data.frame(gtf, header=T)
+    # TODO: For some gtf files, transcript_id does not work!!
```

Comment:
Hm, so this would have to be looked at when the error appears again (i.e. when someone tries a different gtf and it does not work). Or should I create an issue to which people can then respond? I haven't triggered that error since I implemented the gtf stuff...